ffi/src/primitives.rs

```diff
@@ -169,16 +171,43 @@ impl<'itm> IntoFfiSliceRef<'itm> for core::cmp::Ordering {
     }
 }
 
+/// Trait for replacing unsupported types with supported ones when crossing FFI-boundary (for example in case of wasm)
+pub trait PrimitiveRepr {
+    /// Type used to represent [`Self`] when crossing FFI-boundry
+    type Repr;
+}
+
+macro_rules! primitive_repr_impls {
+    ($($source:ty => $target:ty),+ $(,)?) => {$(
+        impl PrimitiveRepr for $source {
+            type Repr = $target;
+        }
+    )+};
+}
+macro_rules! primitive_repr_self_impls {
+    ($($source:ty),+ $(,)?) => {
+        primitive_repr_impls! {
+            $($source => $source),*
+        }
+    };
+}
+
+#[cfg(feature = "wasm")]
+primitive_repr_impls! {u8 => u32, u16 => u32, i8 => i32, i16 => i32}
+#[cfg(not(feature = "wasm"))]
+primitive_repr_self_impls! {u8, u16, i8, i16}
+primitive_repr_self_impls! {u32, u64, u128, i32, i64, i128}
```

Comment:
now you can remove the macros if you follow the previous suggestion

Reply:
I'm not quite understand how your suggestion will help to remove macro.
We still need to implement `WasmRepr` for every supported primitive type, no?
Or you mean that we could remove macro invocation when `wasm` feature is turned off?

Reply:
> Or you mean that we could remove macro invocation when wasm feature is turned off?

Yeah, this is what I meant